### PR TITLE
Author metadata update

### PR DESCRIPTION
**Pull request by jamiehewitt15**
Closes: #724

Changes proposed in this PR:
- Adding author to editable metadata fields
- makes use of https://github.com/oceanprotocol/ocean.js/pull/936

### CODE REVIEW

File: content/pages/edit.json [hunk 1]
```diff
@@ -44,6 +44,13 @@
         "options": ["Forever", "1 day", "1 week", "1 month", "1 year"],
         "sortOptions": false,
         "required": true
+      },
+      {
+        "name": "author",
+        "label": "New Author",
+        "placeholder": "e.g. Mrs McJellyfish",
+        "help": "Give proper attribution for your data set.",
+        "required": true
```
Comment (kremalicious) [inline content/pages/edit.json:53]: here I'm not sure how Formik behaves with this to cover this use case:

> publisher can remove the author as part of _Edit Metadata_ flow

So would clearing the field still make it submittable or would form error be shown?

**Review by kremalicious — CHANGES_REQUESTED** [review]
already merged `main` into this PR. Now we just need to figure out where the `author` should be required or optional

File: content/pages/edit.json (hunk 1 above)
Comment (jamiehewitt15) [reply content/pages/edit.json:53]: Formik won't let you submit an empty form field for the author as it is a required field. The submit button is disabled if you clear the author field

**Comment by jamiehewitt15** [comment]
> already merged `main` into this PR. Now we just need to figure out where the `author` should be required or optional

I would say it should be required. If it is optional users can delete it and leave it blank

**Comment by mihaisc** [comment]
Optional, users need the option to remove it. That what the whole point. This is a gdpr requirement

**Comment by jamiehewitt15** [comment]
@mihaisc I've updated it so that the author is now optional

**Review by mihaisc — APPROVED** [review]

**Review by kremalicious — APPROVED** [review]
awesome, all use cases worked for me

For reference, this is what it looks like after author field has been cleared. Was first thinking to hide it altogether in that case but it's actually nice for transparency as this implies it has been cleared later
Screen Shot 2021-08-03 at 17 04 44

**Comment by jamiehewitt15** [comment]
Thanks, @kremalicious I agree that it's good to keep the author field there, otherwise, users will look for it and be unsure why it's not there. In terms of increasing transparency, we could write _Removed_ instead of leaving a blank space. 

Also, going one step further in terms of transparency. It would also be good if the metadata history could what has been changed. Although this might be a lot of work: 
Metadata history - 2021-08-03_17-37

**Pull request merged**